lib/rubocop/cop/rspec/named_subject.rb

```diff
@@ -59,6 +59,11 @@ class NamedSubject < Cop
         def_node_search :subject_usage, '$(send nil? :subject)'
 
         def on_block(node)
+          RuboCop::RSpec::ExampleGroup.new(node).subjects.each do |subject|
+            next if named_subject?(subject)
+            add_offense(subject)
+          end
+
           return if !rspec_block?(node) || ignored_shared_example?(node)
 
           subject_usage(node) do |subject_node|
```

Comment:
Existing `subject_usage` is sub-optimal since it seems to recursively find `subject` declarations due to `def_node_search`. Should probably be `def_node_matcher` in the first place.